app/helpers/textual_mixins/textual_refresh_status.rb

```diff
@@ -1,20 +1,32 @@
 module TextualMixins::TextualRefreshStatus
   def textual_refresh_status
-    last_refresh_status = @record.last_refresh_status.titleize
-    if @record.last_refresh_date
-      last_refresh_date = time_ago_in_words(@record.last_refresh_date.in_time_zone(Time.zone)).titleize
+    refresh_status(@record)
+  end
+
+  def textual_refresh_date
+    refresh_date(@record)
+  end
+
+  def refresh_status(record)
+    last_refresh_status = record.last_refresh_status&.titleize
+    before_time = Date.today - 2
+    refresh_time = record.last_refresh_date ? record.last_refresh_date.in_time_zone(Time.zone).strftime("%Y-%m-%d").to_date : nil
+    if record.last_refresh_date
+      last_refresh_date = !record.last_refresh_date ? _('Never') : time_ago_in_words(record.last_refresh_date.in_time_zone(Time.zone)).titleize
       last_refresh_status << _(" - %{last_refresh_date} Ago") % {:last_refresh_date => last_refresh_date}
     end
     {
-      :label => _("Last Refresh"),
-      :value => [{:value => last_refresh_status},
-                 {:value => @record.last_refresh_error.try(:truncate, 120)}],
-      :title => @record.last_refresh_error
+      :label      => _("Last Refresh Status"),
+      :value      => [{:value => last_refresh_status},
+                      {:value => record.last_refresh_error.try(:truncate, 120)}],
+      :title      => record.last_refresh_error,
+      :status     => record.last_refresh_status&.titleize,
+      :date_range => !record.last_refresh_date ? false : refresh_time >= before_time # checking if date is within last two days range.
```

Comment:
Rails 6 has added .before? and .after? methods, so I think using that will make this easier to read.  Additionally, let's avoid the double negatives to make this easier to read, as well as use the Rails date helpers.

```suggestion
      :date_range => refresh_time&.before?(2.days.ago)
```

(Note: if nil is allowed here you can use the following, otherwise you'll have to do `refresh_time && refresh_time.before?...`)

Then above,

```diff
-    before_time = Date.today - 2
-    refresh_time = record.last_refresh_date ? record.last_refresh_date.in_time_zone(Time.zone).strftime("%Y-%m-%d").to_date : nil
-    if record.last_refresh_date
-      last_refresh_date = !record.last_refresh_date ? _('Never') : time_ago_in_words(record.last_refresh_date.in_time_zone(Time.zone)).titleize
+    refresh_time = record.last_refresh_date&.in_time_zone(Time.zone)
+    if refresh_time
+      last_refresh_date = refresh_time ? refresh_time.titleize : _('Never')
```